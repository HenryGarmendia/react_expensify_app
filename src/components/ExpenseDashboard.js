import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './expense_summary';

const ExpenseDashboard = () => (
    <div>
        <ExpenseSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboard;