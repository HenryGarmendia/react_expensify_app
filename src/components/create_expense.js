import React from 'react';

const CreateExpense = () => (
    <div>
        This is from ADD component...
    </div>
);

export default CreateExpense;