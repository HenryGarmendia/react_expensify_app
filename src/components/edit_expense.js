import React from 'react';

const EditExpense = (props) => {
   return (
        <div>
            This is from EDIT component width id of {props.match.params.id}.
        </div>
    );
};

export default EditExpense;