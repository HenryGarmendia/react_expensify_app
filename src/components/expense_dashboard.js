import React from 'react';

const ExpenseDashboard = () => (
    <div>
        This is from DASHBOARD component...
    </div>
);

export default ExpenseDashboard;