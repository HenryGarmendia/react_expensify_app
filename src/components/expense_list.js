import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './expense_list_item';
import selectedExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem key={expense.id} {...expense} />
                })
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectedExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);