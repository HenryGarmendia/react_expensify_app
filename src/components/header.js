import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>    
        <h1>Expensify Web-App</h1>
        <NavLink activeClassName="is_active link" exact={true} to="/">Dashboard</NavLink>
        <NavLink activeClassName="is_active link" to="/create">Create Expense</NavLink>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);