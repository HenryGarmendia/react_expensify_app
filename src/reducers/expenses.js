//Expenses Reducer
const expensesReducerDefaultState = [];
const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
        case 'REMOVE_EXPENSE':
            return state.filter(({ id }) => {
                // if this function return true the item will be kep in the array, if return false it will be remove from the array
                return id !== action.id; 
            });
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    // return a brand new object
                    return {
                        ...expense,
                        ...action.updates
                    };
                } else {
                    return expense;
                }
            });
        case 'SET_EXPENSES':
            return action.expenses;
        default:
            return state;
    }
};

export default expensesReducer;