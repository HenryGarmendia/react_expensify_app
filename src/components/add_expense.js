import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './expense_form';
import { addExpense } from '../actions/expenses';

const AddExpense = (props) => (
    <div>
        This is from ADD component...
        <h1>Add Expense</h1>
        <ExpenseForm 
            onSubmit={(expense) => {
                props.dispatch(addExpense(expense));
            }}
        />
    </div>
);

export default connect()(AddExpense);