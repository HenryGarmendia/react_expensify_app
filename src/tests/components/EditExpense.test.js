import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { EditExpense } from '../../components/edit_expense';

// should render EditExpense

// should handle