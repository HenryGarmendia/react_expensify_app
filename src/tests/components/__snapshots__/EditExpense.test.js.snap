// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpense 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        <i
          className="far fa-edit"
        />
         Edit Expense
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <ExpenseForm
      expense={
        Object {
          "amount": 4500,
          "createdAt": 432000000,
          "description": "Credit Card",
          "id": "3",
          "note": "",
        }
      }
      onSubmit={[Function]}
    />
    <button
      className="button button--secondary"
      onClick={[Function]}
    >
      <i
        className="far fa-times-circle"
      />
       Remove
    </button>
  </div>
</div>
`;
